src/ops/fuelup_default.rs

```diff
@@ -0,0 +1,40 @@
+use anyhow::{bail, Result};
+use std::str::FromStr;
+use tracing::info;
+
+use crate::{
+    path::settings_file,
+    settings::SettingsFile,
+    toolchain::{OfficialToolchainDescription, Toolchain},
+};
+
+pub fn default(toolchain: Option<String>) -> Result<()> {
+    let current_toolchain = Toolchain::from_settings()?;
+
+    let toolchain = match toolchain {
+        Some(toolchain) => toolchain,
+        None => {
+            info!("{} (default)", current_toolchain.name);
+            return Ok(());
+        }
+    };
+
+    let mut new_default = Toolchain::from_path(&toolchain)?;
+
+    if let Ok(description) = OfficialToolchainDescription::from_str(&toolchain) {
+        new_default = Toolchain::from_path(&description.to_string())?;
+    }
```

Comment:
Here it looks like we are unnecessarily loading `new_default` twice from different paths in the case that `OfficialToolchainDescription` is `Ok`. Perhaps this could be:

```rust
let new_default = match OfficialToolchainDescription::from_str(&toolchain) {
    Ok(desc) => Toolchain::from_path(&description.to_string())?,
    Err(_) => Toolchain::from_path(&toolchain)?,
};
```

I realise this was copy pasted and previously overlooked, happy to leave for a future PR if preferred.

Reply:
Ah, this is not unnecessarily loaded. The reason is if you run `fuelup default latest`, the `latest` string must be appended with the relevant `TargetTriple` generated from the host machine, which is why `new_default` is mut and we check if we can form an `OfficialToolchainDescription` from the user input string.

I'll add in a short comment to explain this.

Reply:
I'll create a smaller PR for this!

Reply:
I'll just add it to this since I'm making the change below